### ISSUE TITLE
Billing is unavailable

### ISSUE THREAD

**Issue by pouya92** [dooboolab/react-native-iap#728]
### Version of react-native-iap
3.3.9
### Version of react-native
0.59.6
### Platforms you faced the error (IOS or Android or both?)
Android
### Expected behavior
get product list
### Actual behavior
get unhandled promise Rejection, Billing is unavailable. 
### Tested environment (Emulator? Real Device?)
Emulator and Real device
### Steps to reproduce the behavior
at first I should tell its working great on iOS, my problem is only on android, 
Here is my codes.
this is my androidManifest file: 
![Screen Shot 2019-09-23 at 6 24 13 PM](https://user-images.githubusercontent.com/8234096/65436732-75a3a980-de2f-11e9-8845-1d832b82b0a8.png)

here is my setting.Gradle file:
![Screen Shot 2019-09-23 at 6 27 04 PM](https://user-images.githubusercontent.com/8234096/65436964-d501b980-de2f-11e9-8234-a2d7567158b3.png)

here is my dependencies in build.gradle file:
![Screen Shot 2019-09-23 at 6 31 49 PM](https://user-images.githubusercontent.com/8234096/65437393-902a5280-de30-11e9-920c-ebbd2882e7bf.png)

mainApplication.java
![Screen Shot 2019-09-23 at 6 34 09 PM](https://user-images.githubusercontent.com/8234096/65437511-cbc51c80-de30-11e9-8b59-c21673b3f5cc.png)

here is my code in store scene :
First I input the library
```
import * as RNIapX from 'react-native-iap';
import RNIap, {
  purchaseErrorListener,
  purchaseUpdatedListener,
  type ProductPurchase,
  type PurchaseError
} from 'react-native-iap';
```
after that I get my product Ids something like 'gem_30',... as following code:
```
this.itemSkus = Platform.select({
      ios: props.settingProps.allConf.store_pids.app_store,
      android: props.settingProps.allConf.store_pids.play_store
    });
```
in componentDidMount I try to get product list with these codes:
```
_getProductListRequestFromStore = async () => {
    RNIapX.initConnection()
    try {
      const products: Product[] = await RNIapX.getProducts(this.itemSkus);     
    } catch(err) { 
     console.warn('error',err)
    }
  }
```
in above code I will catch this error :
error{"farmeToPop" : 1, "nativeStackAndroid":[], "userInfo":null,...}

and here is my listener function:
```
_createIAPListenner = () => {
    //IAP
    //console.log('in create listenner func')    
    this.purchaseUpdateSubscription = purchaseUpdatedListener((purchase: ProductPurchase) => {
      //this.setState({loading: true})
      //console.log('purchaseUpdatedListener', purchase);
      const receipt = purchase.transactionReceipt;
      if (receipt) {
        //Call server Api
        this._sendPurchaseDataToServer(receipt, purchase);
        //Call server Api        
      }
    });
    this.purchaseErrorSubscription = purchaseErrorListener((error: PurchaseError) => {
      this.setState({loading: false})
      //console.warn('purchaseErrorListener', error);
    });
    //IAP
  }
```
this code works greate on ios but keeps getting me this error on android.
Also, I published my app and makes tester group and attach it to alpha version to test purchase flow. 
please help me to find where I'm going wrong.

thanks for your great library and efforts.

**Comment by abaddonGIT** [comment]
Check what you logged in google play service in your emulator.

**Comment by pouya92** [comment]
I checked and I'm logged in, I'm using real device to test. my account is the one I add in google play console testers account

**Comment by pouya92** [comment]
in listener I get unhandled promise error, Billing is not available:
```
this.purchaseUpdateSubscription = purchaseUpdatedListener((purchase: ProductPurchase) => {
      //console.log('purchaseUpdatedListener', purchase);
      const receipt = purchase.transactionReceipt;
      if (receipt) {
        //Call server Api
        this._sendPurchaseDataToServer(receipt, purchase);
        //Call server Api        
      }
    });
```

**Comment by hyochan** [comment]
Try chaning `RNIapX.initConnection()` to `await RNIapX.initConnection()`. If this doesn't still work, please check what `initConnection is returning`.

**Comment by pouya92** [comment]
Hi, thanks for answer, I did as you told but nothings changed.
```
_getProductListRequestFromStore = async () => {
    await RNIapX.initConnection()
   console.warn('Hello');
    try {
      const products: Product[] = await RNIapX.getProducts(this.itemSkus);
      console.warn("store products",products)    
    } catch(err) {    
      console.warn('err', err); // standardized err.code and err.message available
    }
  }
```
when I add await, It seems it wont do anything. Not even show me error anymore, or even hello warning, it seems it wont go any further.

but if I dont use await I get this warning in RNIapX.getProducts(this.itemSkus) :
![WhatsApp Image 2019-09-25 at 4 53 56 PM](https://user-images.githubusercontent.com/8234096/65604960-2b4c3500-dfb5-11e9-95e9-ed32b4d0b263.jpeg)


In Lisstenner part I also add await RNIapX.initConnection() so I dont get Unhandled Prommise warning anymore but before that I always got this error :
![warning](https://user-images.githubusercontent.com/8234096/65605149-7e25ec80-dfb5-11e9-88d0-7b5fb045ab6f.jpeg)


I think its related to my account, on iOS it's working like a charm, on android I have this issue.
I checked I'm signed in google play.
Do I need to add a payment method to my account or dont?
